Google.PowerShell/Compute/GceHealthCheckCmdlets.cs

```diff
@@ -0,0 +1,545 @@
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpsObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpsHealthCheck HttpsObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Object describing a new http health check.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByHttpObject, Mandatory = true,
+            Position = 0, ValueFromPipeline = true)]
+        public HttpHealthCheck HttpObject { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (Https || HttpsObject != null)
+            {
+                HttpsHealthCheck body = HttpsObject ?? new HttpsHealthCheck
+                {
+                    Name = Name,
+                    Description = Description,
+                    Host = HostHeader,
+                    Port = Port,
+                    RequestPath = RequestPath,
+                    CheckIntervalSec = (int?)CheckInterval?.TotalSeconds,
+                    TimeoutSec = (int?)Timeout?.TotalSeconds,
+                    HealthyThreshold = HealthyThreshold,
+                    UnhealthyThreshold = UnhealthyThreshold
+                };
+                HttpsHealthChecksResource.InsertRequest request =
+                    Service.HttpsHealthChecks.Insert(body, Project);
+                Operation operation = request.Execute();
+                AddGlobalOperation(Project, operation, () =>
+                {
+                    WriteObject(Service.HttpsHealthChecks.Get(Project, body.Name).Execute());
+                });
+            }
+            else
+            {
+                HttpHealthCheck body = HttpObject ?? new HttpHealthCheck
+                {
+                    Name = Name,
+                    Description = Description,
+                    Host = HostHeader,
+                    Port = Port,
+                    RequestPath = RequestPath,
+                    CheckIntervalSec = (int?)CheckInterval?.TotalSeconds,
+                    TimeoutSec = (int?)Timeout?.TotalSeconds,
+                    HealthyThreshold = HealthyThreshold,
+                    UnhealthyThreshold = UnhealthyThreshold
+                };
+                HttpHealthChecksResource.InsertRequest request =
+                    Service.HttpHealthChecks.Insert(body, Project);
+                Operation operation = request.Execute();
+                AddGlobalOperation(Project, operation, () =>
+                {
+                    WriteObject(Service.HttpHealthChecks.Get(Project, body.Name).Execute());
+                });
+            }
+        }
+    }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Gets Google Compute Engine health checks.
+    /// </para>
+    /// <para type="description">
+    /// Gets Google Compute Engine health checks.
+    /// This cmdlet can be used to retrieve both http and https health checks.
```

Comment:
Mention it can get all checks for a project, or a subset based on the name, etc.